### ISSUE TITLE
Missing clearUser method on react-native

### ISSUE THREAD

**Issue by ghsdh3409** [bugsnag/bugsnag-js#1259]
### Describe the bug
* As an official reference [document](https://docs.bugsnag.com/platforms/react-native/react-native/legacy/configuration-options/#clearuser), there is `clearUser` method. But, after I upgrading v7, there is no `clearUser`.
* Is this breaking change? If not, what should I use instead of `clearUser`?

### Steps to reproduce
1. Use `Bugsnag.clearUser()`

### Environment
* Bugsnag version: 7.6
* React Native: 0.63.4
* Device (e.g. iphonex): Android and iOS

### Error screenshot
<img width="323" alt="스크린샷 2021-01-22 오후 3 58 41" src="https://user-images.githubusercontent.com/6203798/105458048-2ee7ed00-5ccb-11eb-8cf8-d84acc2049c0.png">

**Comment by xander-jones** [comment]
Hey @ghsdh3409, the docs you've linked to are our legacy documents for the deprecated `bugsnag-react-native` package. The `clearUser()` method does not exist in the new React Native package.

In Bugsnag for React Native using `bugsnag-js` v7.3.0+, you can clear the user data though by calling `Bugsnag.setUser();` without any arguments in the JS layer.

**Comment by jdferreira-biai** [comment]
@xljones Thanks!

Btw, that should be added in the docs. I spent some time reading the docs looking for how to clear the user data.

**Comment by hannah-smartbear** [comment]
Hi @jdferreira-biai,

You can find information on how to clear user data using `setUser` in our docs [here](https://docs.bugsnag.com/platforms/javascript/customizing-error-reports/#setuser:~:text=Nag%27)-,null,be%20used%20as%20a%20parameter%20value%20to%20clear%20the%20user%20attribute.,-severity). Please let us know if you have any further questions!

**Comment by jdferreira-biai** [comment]
> Hi @jdferreira-biai,
> 
> You can find information on how to clear user data using `setUser` in our docs [here](https://docs.bugsnag.com/platforms/javascript/customizing-error-reports/#setuser). Please let us know if you have any further questions!

@hannah-smartbear There is an issue. When I try to send `null` as value for the `setUser` function I get this type error:

```
Argument of type 'null' is not assignable to parameter of type 'string | undefined'. ts(2345)
```

### Example:
```
import Bugsnag from '@bugsnag/js'

Bugsnag.setUser(null) // here i get the type error
```

I'm using `@bugsnag/js@8.1.2`.

**Comment by mclack** [comment]
Hi @jdferreira-biai 

I just wanted to update you here, as we recently release v8.2.0 of `bugsnag-js`, which corrects the `setUser` TS interface to allow for `null` arguments: https://github.com/bugsnag/bugsnag-js/releases/tag/v8.2.0

Please let us know if you have any further questions or issues. You can open a ticket with us directly by contacting support@bugsnag.com.

**Comment by jdferreira-biai** [comment]
> Hi [@jdferreira-biai](https://github.com/jdferreira-biai)
> 
> I just wanted to update you here, as we recently released v8.2.0 of `bugsnag-js`, which corrects the `setUser` TS interface to allow for `null` arguments: https://github.com/bugsnag/bugsnag-js/releases/tag/v8.2.0
> 
> Please let us know if you have any further questions or issues. You can open a ticket with us directly by contacting [support@bugsnag.com](mailto:support@bugsnag.com).

Nice! Thanks for working on that. 👍🏻

So if I want to unset or clear the user I should do the following?

```
Bugsnag.setUser(null, null, null)
```

Also, it's there any difference if I do instead this?
```
Bugsnag.setUser()
```
(that's what I've been doing until now.

**Comment by mclack** [comment]
Hi @jdferreira-biai 

When calling `Bugsnag.setUser()` with no arguments, this should reset the user metadata to the default, which means that BugSnag will still attempt to capture the user's IP address, and this will be stored in the event as `user.id`.

However, when explicitly passing `null` arguments like `Bugsnag.setUser(null, null, null)`, this should completely wipe all `user` metadata, meaning nothing would be reported along with the event, and no `USER` tab should appear.

Hopefully that makes sense, but please let us know if you have any other questions.
